src/utilities.cc

```diff
@@ -296,13 +300,17 @@ const string& MyUserName() {
   return g_my_user_name;
 }
 static void MyUserNameInitializer() {
-  // TODO(hamaji): Probably this is not portable.
+  int err = 0;
+  char user[1024];
 #if defined(OS_WINDOWS)
-  const char* user = getenv("USERNAME");
+  DWORD bufsize = sizeof(user);
```

Comment:
This does not look right.

Reply:
`user` could be `NULL`.

Reply:
I apologize for the delay.
Please look at the complete diff.

`user` was originally of type `const char*`.  I made changes so it is now a string (as required by the APIs) - `char user[1024];`.

So `sizeof(user)` now actually returns `sizeof(char) * 1024`.